corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -447,6 +450,69 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
             }
         },
 
+        columnStyle: function () {
+            let self = this;
+            let mapColumn = "";
+            if (self.showMap && !self.hasNoItems) {
+                mapColumn = "[map] 300px";
+            }
+            return "display: grid;grid-template-columns: [tiles] auto " + mapColumn
+                + " ;grid-template-rows: auto";
+        },
+
+        addAddressPin: function (geocoder, addressMap, headers, model, addressIndex) {
+            let address = model.attributes.data[addressIndex];
+            geocoder.query(address, function (err, data) {
+                if (err === null) {
+                    let popupText = "";
+                    headers.forEach((header, index) => {
+                        if (header) {
+                            popupText += "<b>" + header + ":</b> " + model.attributes.data[index] + "<br>";
```

Comment:
Since it seems `bindPopup` accepts HTML, the header & data should be sanitized with `DOMPurify`.

Reply:
(this is my only blocking comment)